lib/active_merchant/billing/gateways/sum_up.rb

```diff
@@ -0,0 +1,192 @@
+        transcript.
+          gsub(%r((Authorization: Bearer )\w+), '\1[FILTERED]').
+          gsub(%r(("pay_to_email\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("number\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("cvv\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]')
+      end
+
+      private
+
+      def upsert_checkout(money, payment, options = {})
+        if options[:checkout_id].present?
+          find_checkout(options[:checkout_id])
+        else
+          create_checkout(money, payment, options)
+        end
+      end
+
+      def find_checkout(checkout_id)
+        commit('checkouts/' + checkout_id, {}, :get)
+      end
+
+      def create_checkout(money, payment, options)
+        post = {}
+
+        add_merchant_data(post, options)
+        add_invoice(post, money, options)
+        add_address(post, payment, options)
+        add_customer_data(post, payment, options)
+
+        commit('checkouts', post)
+      end
+
+      def complete_checkout(checkout_id, payment, options = {})
+        post = {}
+
+        add_payment(post, payment, options)
+
+        commit('checkouts/' + checkout_id, post, :put)
+      end
+
+      def add_customer_data(post, payment, options)
+        post[:customer_id]            = options[:customer_id]
+        personal_details              = post[:personal_details] ||= {}
+        personal_details[:email]      = options[:email]
+        personal_details[:first_name] = payment&.first_name
+        personal_details[:last_name]  = payment&.last_name
+        personal_details[:tax_id]     = options[:tax_id]
+      end
+
+      def add_merchant_data(post, options)
+        # Required field: pay_to_email
+        # Description: Email address of the merchant (in this case Spreedly)
+        # to whom the payment is made.
+        post[:pay_to_email] = @options[:pay_to_email]
+        post[:redirect_url] = options[:redirect_url]
+        post[:return_url]   = options[:return_url]
+      end
+
+      def add_address(post, creditcard, options)
+        personal_details      = post[:personal_details] ||= {}
+        address               = personal_details[:address] = {}
+        address[:city]        = options[:billing_address][:city]
+        address[:state]       = options[:billing_address][:state]
+        address[:country]     = options[:billing_address][:country]
+        address[:line_1]      = options[:billing_address][:address1]
+        address[:postal_code] = options[:billing_address][:zip]
+      end
+
+      def add_invoice(post, money, options)
+        post[:checkout_reference] = options[:checkout_reference]
+        post[:amount]             = amount(money)
+        post[:currency]           = (options[:currency] || currency(money))
+        post[:description]        = options[:description]
+      end
+
+      def add_payment(post, payment, options)
+        post[:payment_type] = options[:payment_type] if PAYMENT_TYPES.include?(options[:payment_type])
```

Comment:
❓ Question:

Wouldn't be possible to identify the 'payment_type' from the card brand in conjunction with the payment method class ?

Reply:
I looked in the code and I don't see a way to do it, maybe in another instance it could work, not in this gem

Reply:
One idea => 

Well we could confirm with Charlotte/Nick but instead of thinking that the merchant is going to send us a GSF with the proper value for `options[:payment_type]` we could just start with the type 'card' of have it as a fallback if no value is available on `options[:payment_type]`

Reply:
Perfect, I'm going to ask and for now I'll leave `card` as the default value